### PR TITLE
Fix: clippy warning in rust 1.62.1

### PR DESCRIPTION
**Pull request by dlhxzb**
Just resolve warnings in cargo clippy and build
> $ rustc -V
rustc 1.62.1 (e092d0b6b 2022-07-16)
  
@rustdesk PTAL

### CODE REVIEW

**Comment by paspo** [comment]
I just tried and every warning seems solved.
Just remember a touch of `cargo fmt`.

FYI on rust versions:

- ubuntu 20.04 uses rust 1.59.0
- ubuntu 22.04 uses rust 1.59.0
- our automatic build (via girhub actions) is [ubuntu-22.04](https://github.com/actions/virtual-environments)

So this is not urgent, but it's welcome.

LGTM

**Comment by dlhxzb** [comment]
Of course I had ran `cargo fmt`

**Comment by paspo** [comment]
> our automatic build (via girhub actions) is [ubuntu-22.04](https://github.com/actions/virtual-environments)

which now uses rust 1.62.1, as stated [here](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md)

Now this PR is gaining priority.

**Comment by paspo** [comment]
@rustdesk PTAL

**Comment by rustdesk** [comment]
@dlhxzb sorry for ignoring this PR and introduced some conflicts. Could you help solve the conflicts?

File: src/rendezvous_server.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ const ADDR_127: IpAddr = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
 
 #[derive(Clone, Debug)]
 enum Data {
-    Msg(RendezvousMessage, SocketAddr),
+    Msg(Box<RendezvousMessage>, SocketAddr),
```
Comment (rustdesk) [inline src/rendezvous_server.rs:43]: Just wondering, box is more efficient here?

**Comment by rustdesk** [comment]
Great job!

File: src/rendezvous_server.rs (hunk 1 above)
Comment (dlhxzb) [reply src/rendezvous_server.rs:43]: https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant  
IMO when alloc enum in memory, the size = tag+max variant, so better to keep variants have less size differences

**Comment by rustdesk** [comment]
@dlhxzb Thank you so much! do you have interest to fix clippy in rustdesk repo? We can do it part by part.

**Comment by dlhxzb** [comment]
@rustdesk I will join a new company next month, so maybe a bit busy recently. I will do it if I have time

**Comment by rustdesk** [comment]
@dlhxzb Congratulations and Thanks. Take time, :)

**Pull request merged**